### PR TITLE
Add Update tenant minioInstance api

### PR DESCRIPTION
**Pull request by cesnietor**
This patches the minio image of the minioInstance.

api: PUT `/api/v1/tenants/{tenant-name}`
payload:
```
{
   "image" : "minio/minio:RELEASE.2020-06-03T22-13-49Z"
}
``` 
Operator should take care of the rest by patching the minioInstances and restarting the pods

### CODE REVIEW

File: restapi/operator_client.go [hunk 1]
```diff
@@ -42,3 +46,13 @@ type operatorClient struct {
 func (c *operatorClient) MinIOInstanceDelete(ctx context.Context, currentNamespace string, instanceName string, options metav1.DeleteOptions) error {
 	return c.client.OperatorV1().MinIOInstances(currentNamespace).Delete(ctx, instanceName, options)
 }
+
+// MinIOInstanceGet implements the minio instance get action from minio-operator
+func (c *operatorClient) MinIOInstanceGet(ctx context.Context, currentNamespace string, instanceName string, options metav1.GetOptions) (*v1.MinIOInstance, error) {
```
Comment (dvaldivia) [inline restapi/operator_client.go:51]: Maybe we don't have to call it current name space anymore

File: restapi/tenants.go [hunk 2]
```diff
@@ -354,3 +368,52 @@ func getTenantCreatedResponse(token string, params admin_api.CreateTenantParams)
...
+	minInst, err := operatorClient.MinIOInstanceGet(ctx, nameSpace, params.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// if image to update is empty we'll use the latest image by default
+	if strings.TrimSpace(imageToUpdate) != "" {
+		minInst.Spec.Image = params.Body.Image
+	} else {
+		minInst.Spec.Image = "minio/minio:latest"
```
Comment (dvaldivia) [inline restapi/tenants.go]: Instead of using the latest tag please use the function we have to retrieve the latest release

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: restapi/operator_client.go (hunk 1 above)
Comment (cesnietor) [reply restapi/operator_client.go:51]: that work for now but that will change on the pr for namescape scope

File: restapi/tenants.go (hunk 2 above)
Comment (dvaldivia) [reply restapi/tenants.go]: `getLatestMinIOImage()`

Comment (cesnietor) [reply restapi/tenants.go]: done @dvaldivia

**Review by dvaldivia — APPROVED** [review]

**Review by bexsoft — COMMENTED** [review]
LGTM

**Pull request merged**